sbi/inference/posteriors/ratio_based_posterior.py

```diff
@@ -330,7 +334,9 @@ def np_potential(self, theta: np.array) -> ScalarFloat:
         theta = ensure_theta_batched(theta)
         x = ensure_x_batched(self.x)
 
-        log_ratio = self.classifier(torch.cat((theta, x), dim=1).reshape(1, -1))
+        log_ratio = self.classifier(
+            torch.cat((theta.to(x.device), x), dim=1).reshape(1, -1)
+        ).cpu()
```

Comment:
So, mcmc sampling always happens on `cpu`? Why?

Reply:
Same as above, the computation is happening on the device, therefore `theta.to(device)` (`x` is there already), but the prior is on CPU, therefore `.cpu()`.